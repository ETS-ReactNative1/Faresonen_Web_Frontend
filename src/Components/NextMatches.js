import React, { Component } from 'react';
import { List, Item, Divider } from 'semantic-ui-react';
import axios from 'axios';
import MatchInfo from './MatchInfo';

class NextMatches extends Component {
	
	constructor(props) {
		super(props);
		this.state = {
			matches: [],
			loading: true
		}
	}

	

	componentDidMount() {

	}

	componentWillReceiveProps(nextProps) {
		console.log(nextProps);
		this.setState({
			matches: nextProps.matches,
			loading: false
		});
	}

	

	render() {
		if(this.state.loading) {
			return <p>Henter kamper</p>
		}
		const nextMatches = this.state.matches.map((m) => {
			return (
				<List.Item key={m.id}>
					<MatchInfo match={m}>
					</MatchInfo>
					<Divider />
				</List.Item>
				)
		});
		return (
			<div>
				<List>
					{ nextMatches }
				</List>
			</div>
			)
	}
}

export default NextMatches;