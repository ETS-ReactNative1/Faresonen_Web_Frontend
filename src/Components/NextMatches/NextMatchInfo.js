import React, {Component} from "react";
import PropTypes from "prop-types";
import {Table} from "semantic-ui-react";
import axios from "axios";
import tools from "../../Tools/tools";

class NextMatchInfo extends Component {
  constructor(props) {
    super(props);
    this.state = {
      matchInfo: props.matchInfo,
      nextRoundNumber: parseInt(props.nextRoundNumber, 10) + 1,
      stadium: "",
      channel: "",
      date: tools.getDate(props.matchInfo.starttime),
      time: tools.getTime(props.matchInfo.starttime),
    };

    axios.all([tools.getChannelAndStadium(props.matchInfo)]).then(
      axios.spread((data) => {
        if (data.channel) {
          data.channel.then(channel => this.setState({channel: channel.data.name}));
        } else {
          this.setState({
            channel: "Kanal ikke klar",
          });
        }

        if (data.stadium) {
          data.stadium.then(stadium => this.setState({stadium: stadium.data.name}));
        } else {
          this.setState({
            stadium: "Stadion klar",
          });
        }
      }),
    );
  }

  render() {
    return (
      <Table.Row>
        <Table.Cell>{this.state.date}</Table.Cell>
        <Table.Cell>{this.state.time}</Table.Cell>
        <Table.Cell>
          {this.state.nextRoundNumber}
. runde
        </Table.Cell>
        <Table.Cell>{this.state.matchInfo.name}</Table.Cell>
        <Table.Cell>{this.state.stadium}</Table.Cell>
        <Table.Cell>{this.state.channel}</Table.Cell>
      </Table.Row>
    );
  }
}

NextMatchInfo.propTypes = {
  matchInfo: PropTypes.object.isRequired,
  nextRoundNumber: PropTypes.number.isRequired,
};

export default NextMatchInfo;
