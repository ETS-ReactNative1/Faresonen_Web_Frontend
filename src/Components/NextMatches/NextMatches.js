import React from "react";
import PropTypes from "prop-types";
import { Table, Segment, Dimmer, Loader, Message } from "semantic-ui-react";
import NextMatchInfo from "./NextMatchInfo";

const NextMatches = ({ matches, nextRoundNumber, loading }) => {
  const nextMatchInfo = matches.map(m => (
    <NextMatchInfo key={m.id} matchInfo={m} nextRoundNumber={nextRoundNumber} />
  ));

  if (nextMatchInfo.length <= 0 && !loading) {
    return (
      <Message className="no-print" info>
        Kamper for neste runde er ikke klar, eller så holder du på å lage lefsen
        for sesongens siste runde.
      </Message>
    );
  }

  return (
    <Segment>
      <Dimmer active={loading}>
        <Loader>Henter neste runde</Loader>
      </Dimmer>
      <h1>Neste runde</h1>
      <Table striped>
        <Table.Header>
          <Table.Row>
            <Table.HeaderCell>Dato</Table.HeaderCell>
            <Table.HeaderCell>Kl.</Table.HeaderCell>
            <Table.HeaderCell>Runde</Table.HeaderCell>
            <Table.HeaderCell>Kamp</Table.HeaderCell>
            <Table.HeaderCell>Arena</Table.HeaderCell>
            <Table.HeaderCell>Kanal</Table.HeaderCell>
          </Table.Row>
        </Table.Header>
        <Table.Body>{nextMatchInfo}</Table.Body>
      </Table>
    </Segment>
  );
};

NextMatches.propTypes = {
  matches: PropTypes.arrayOf(PropTypes.shape({})).isRequired,
  nextRoundNumber: PropTypes.number.isRequired,
  loading: PropTypes.bool.isRequired
};

export default NextMatches;
