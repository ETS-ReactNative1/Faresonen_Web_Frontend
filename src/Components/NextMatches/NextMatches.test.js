import React from "react";
import { shallow } from "enzyme";
import renderer from "react-test-renderer";
import NextMatches from "./NextMatches";

describe("<NextMatches />", () => {
  it("Should render properly", () => {
    let elem = renderer.create(
      <NextMatches matches={[]} nextRoundNumber={2} loading={false} />
    );
    expect(elem).toMatchSnapshot();

    elem = renderer.create(
      <NextMatches matches={[{ id: 1 }]} nextRoundNumber={2} loading={false} />
    );
    expect(elem).toMatchSnapshot();
  });
});
