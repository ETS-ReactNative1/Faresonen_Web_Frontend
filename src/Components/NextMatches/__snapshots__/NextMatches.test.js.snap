// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<NextMatches /> Should render properly 1`] = `
<div
  className="ui info message no-print"
>
  Kamper for neste runde er ikke klar, eller så holder du på å lage lefsen for sesongens siste runde.
</div>
`;

exports[`<NextMatches /> Should render properly 2`] = `
<div
  className="ui segment"
>
  <div
    className="ui dimmer"
    onClick={[Function]}
  >
    <div
      className="content"
    >
      <div
        className="center"
      >
        <div
          className="ui text loader"
        >
          Henter neste runde
        </div>
      </div>
    </div>
  </div>
  <h1>
    Neste runde
  </h1>
  <table
    className="ui striped table"
  >
    <thead
      className=""
    >
      <tr
        className=""
      >
        <th
          className=""
        >
          Dato
        </th>
        <th
          className=""
        >
          Kl.
        </th>
        <th
          className=""
        >
          Runde
        </th>
        <th
          className=""
        >
          Kamp
        </th>
        <th
          className=""
        >
          Arena
        </th>
        <th
          className=""
        >
          Kanal
        </th>
      </tr>
    </thead>
    <tbody
      className=""
    >
      <tr
        className=""
      >
        <td
          className=""
        >
          Invalid Date
        </td>
        <td
          className=""
        >
          Invalid Date
        </td>
        <td
          className=""
        >
          3
          . runde
        </td>
        <td
          className=""
        />
        <td
          className=""
        >
          
        </td>
        <td
          className=""
        >
          
        </td>
      </tr>
    </tbody>
  </table>
</div>
`;
